### PR TITLE
Add mikoto board

### PR DESCRIPTION
**Pull request by mrninhvn**
From https://github.com/zhiayang/mikoto

### CODE REVIEW

File: src/boards/mikoto/board.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+//--------------------------------------------------------------------+
+#define BLEDIS_MANUFACTURER  "zhiayang ft mrninhvn"
+#define BLEDIS_MODEL         "mikoto"
+
+//--------------------------------------------------------------------+
+// USB
+//--------------------------------------------------------------------+
+#define USB_DESC_VID           0x7A79
+#define USB_DESC_UF2_PID       0x0520
+#define USB_DESC_CDC_ONLY_PID  0x0520
```
Comment (hathach) [inline src/boards/mikoto/board.h:58]: could you confirm this is the VID/PID assigned specifically for this board.

**Review by hathach — COMMENTED** [review]
thank you for your PR. Could you confirm the VID/PID as review comment.

File: src/boards/mikoto/board.h (hunk 1 above)
Comment (mrninhvn) [reply src/boards/mikoto/board.h:58]: > could you confirm this is the VID/PID assigned specifically for this board.

7A79 is "zy" (zhiayang), I haven't seen this duplicated anywhere, but do you have any suggestions to be sure? Thank you.

**Review by hathach — CHANGES_REQUESTED** [review]
you cannot randomly pick an vid/pid. These ids are uniquely assigned by vendor for each product and this repo only accept one with valid vid/pid. You should try to apply for an vid/pid from other donor.

**Comment by mrninhvn** [comment]
> you cannot randomly pick an vid/pid. These ids are uniquely assigned by vendor for each product and this repo only accept one with valid vid/pid. You should try to apply for an vid/pid from other donor.

hmm, is there an easy way for me to get one? thank you.

**Comment by hathach** [comment]
you could try to apply for one https://pid.codes/

**Comment by zhiayang** [comment]
I've got a PID now: 1209 / 7a01. https://github.com/pidcodes/pidcodes.github.com/pull/701

**Comment by mrninhvn** [comment]
> I've got a PID now: 1209 / 7a01. [pidcodes/pidcodes.github.com#701](https://github.com/pidcodes/pidcodes.github.com/pull/701)

Thanks @zhiayang , I changed that.

File: src/boards/mikoto/board.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+//--------------------------------------------------------------------+
+// USB
+//--------------------------------------------------------------------+
+#define USB_DESC_VID           0x1209
+#define USB_DESC_UF2_PID       0x7A01
+#define USB_DESC_CDC_ONLY_PID  0x7A01
+
+#define UF2_PRODUCT_NAME  "Mikoto nRF52840"
+#define UF2_VOLUME_LABEL  "MIKOTO-BOOT"
+#define UF2_BOARD_ID      "nRF52840-mikoto"
```
Comment (hathach) [inline src/boards/mikoto/board.h:62]: I am not picky but board id must follow uf2 spec https://github.com/microsoft/uf2#files-exposed-by-bootloaders . You need to add the board revision as well

```
The Board-ID field is machine-readable and consists of a number of dash-separated tokens. The first token is the CPU type, second is the board type, and third is the board revision. More tokens can be also added.
```

**Review by hathach — CHANGES_REQUESTED** [review]
everything look good, only a minor change request to correct the board id format

File: src/boards/mikoto/board.h (hunk 2 above)
Comment (mrninhvn) [reply src/boards/mikoto/board.h]: > 

Thank you, I changed that.

**Review by hathach — COMMENTED** [review]
look good, it is ready to merge, though for board binary to be part of the release, the board need to be added to the ci as well https://github.com/adafruit/Adafruit_nRF52_Bootloader/blob/master/.github/workflows/githubci.yml#L27

**Review by hathach — APPROVED** [review]
perfect now

**Pull request merged**